enterprise/frontend/src/embedding-sdk/hooks/public/use-metabase-auth-status.ts

```diff
@@ -1,4 +1,19 @@
+import { useContext } from "react";
+import { ReactReduxContext } from "react-redux";
+import { t } from "ttag";
+
 import { useSdkSelector } from "embedding-sdk/store";
 import { getLoginStatus } from "embedding-sdk/store/selectors";
 
-export const useMetabaseAuthStatus = () => useSdkSelector(getLoginStatus);
+// eslint-disable-next-line no-literal-metabase-strings -- this string only shows in the console.
+export const USE_OUTSIDE_OF_CONTEXT_MESSAGE = t`The useMetabaseAuthStatus hook must be used within a component wrapped by the MetabaseProvider`;
+
+export function useMetabaseAuthStatus() {
+  const context = useContext(ReactReduxContext);
+
+  if (!context) {
+    throw new Error(USE_OUTSIDE_OF_CONTEXT_MESSAGE);
+  }
+
+  return useSdkSelector(getLoginStatus);
```

Comment:
Should we make useSdkSelector more than an alias so we can add this logic there? I'm afraid we'll forget to add this check to future selectors/hooks we may export

Reply:
@npretto good point, I'll add it there then